va/validation-authority.go

```diff
@@ -518,7 +544,11 @@ func (va *ValidationAuthorityImpl) validate(ctx context.Context, authz core.Auth
 func (va *ValidationAuthorityImpl) validateChallengeAndCAA(ctx context.Context, identifier core.AcmeIdentifier, challenge core.Challenge) ([]core.ValidationRecord, *probs.ProblemDetails) {
 	ch := make(chan *probs.ProblemDetails, 1)
 	go func() {
-		ch <- va.checkCAA(ctx, identifier)
+		if va.caaClient == nil {
+			ch <- va.checkCAA(ctx, identifier)
+			return
```

Comment:
Because this is a implementation switch, you should use an else instead of a return.